Duct insulation air film

nathancoliver opened the pull request:
I would like to start a conversation about updating uninsulated and insulated duct supply and return R-values, per Std 301 changes. Below is a summary of the changes.

base.pxt
- Changed default air film R-values from 1.5 to 1.7. This defines uninsulated duct effective R-values.
- Created air film and insulation R-value variables for supply and return ducts.

&lt;case name&gt;.pxv
- Set insulation R-values to 0.
- Set air film R-values equal to Std 301 effective supply and return duct insulation R-values.

Let me know if there is anything that should be updated.

nathancoliver committed:
Update duct air film default R-values, and define air film and insulation R-value variables for both supply and return ducts.

nathancoliver committed:
Update supply and return duct R-values.

nealkruis commented:
@nathancoliver, can you make this DRYer? Since the air film is calculated, it probably shouldn't be an input parameter in the template.

nathancoliver commented:
@nealkruis do you think I could change the .pxv files so that the duct air film is equal to 6, and change the base.pxt file to include an if-else statement, where the if statement uses the default air film, and the else statement uses the equation to calculate the air film?

nathancoliver committed:
Add definition to calculate duct air film r-value from nominal duct insulation r-value.

nathancoliver committed:
Remove duct air film r-values and update nominal duct insulation r-values.

nathancoliver commented:
@nealkruis I updated the branch per our conversation. The changes should be self-explanatory, but for our records, here is an overview of the changes.
- Removed duct air film r-values from .pxv files.
- Updated duct insulation r-values to reflect the nominal insulation r-value.
- Added a definition in base.pxt to calculate an air film r-value from the nominal duct insulation that will result in an effective insulation r-value as specified in Standard 301.
- The definition is only called when the duct is insulated.

nealkruis commented on base.pxt:
```diff
@@ -45,8 +45,10 @@
-  parameter "duct_insulation_r_value", :default=>0 # R-value of duct insulation without air film
-  parameter "duct_air_film_r_value", :default=>1.5 # R-value of exterior air film for duct (or duct insulation)
+  parameter "duct_insulation_r_value_supply", :default=>0 # R-value of duct insulation without air film
+  parameter "duct_air_film_r_value_supply", :default=>1.7 # R-value of exterior air film for duct (or duct insulation)
```
This parameter should be removed and always calculated in the template itself (including the logic for R=0).

nathancoliver replied:
Fixed.

nealkruis commented on base.pxt:
```diff
@@ -45,8 +45,10 @@
+  parameter "duct_insulation_r_value_supply", :default=>0 # R-value of duct insulation without air film
+  parameter "duct_air_film_r_value_supply", :default=>1.7 # R-value of exterior air film for duct (or duct insulation)
+  parameter "duct_insulation_r_value_return", :default=>0 # R-value of duct insulation without air film
+  parameter "duct_air_film_r_value_return", :default=>1.7 # R-value of exterior air film for duct (or duct insulation)
```
Ditto here.

nathancoliver replied:
Fixed.

nealkruis commented on base.pxt:
```diff
@@ -647,27 +657,33 @@ window_area = 0.0
     rsVFPerTon = <%= fan_flow_per_cap %>
 
   <% if supply_duct_area > 0 %>
+    <% if duct_insulation_r_value_supply > 0 %>
```
Handle the R=0 case in the function above (don't rely on parameter default for the air film since it shouldn't be an input parameter).

nathancoliver replied:
Fixed.

nealkruis commented on base.pxt:
```diff
@@ -647,27 +657,33 @@ window_area = 0.0
   <% end %>
 
   <% if return_duct_area > 0 %>
+    <% if duct_insulation_r_value_return > 0 %>
```
Ditto here.

nathancoliver replied:
Fixed.

nealkruis commented on base.pxt:
```diff
@@ -75,6 +77,14 @@ def c_f(c)
+  elsif duct_type == "RETURN"
+    return 2.0388 - 0.2947 * duct_insulation_r_value
+  end
+end
```
Did you test what happens if the air film is less than 1?

nathancoliver replied:
I performed some tests, and it appears that negative air films close to zero produce unusual and unexpected heating and cooling energy use results. Other air film values, including positive numbers between 0 and 1 and negative numbers less than 1, do not produce strange results.

nathancoliver committed:
Perform duct air film calculation inside function when duct is uninsulated.

nathancoliver commented:
@nealkruis I moved the air film calculation when the duct is uninsulated into the 'calculate air film r-value' function.

nathancoliver committed:
Merge remote-tracking branch 'origin/master' into duct-insulation-air-film

nathancoliver committed:
Update effective duct R-values according to 301-2022 Add. C.

nathancoliver committed:
Merge remote-tracking branch 'origin/master' into duct-insulation-air-film
